src/hotspot/share/asm/register.hpp

```diff
@@ -45,8 +45,29 @@ class AbstractRegisterImpl {
   int value() const                              { return (int)(intx)this; }
 };
 
+#define AS_REGISTER(type,name)         ((type)name##_##type##EnumValue)
+
+#ifndef USE_EXTERN_CONST_FOR_REGISTER_DEFINITIONS
```

Comment:
I can see this conditionalization being useful for development and testing of this change, but why keep it and the old, now dead, code?

Reply:
The old code was a workaround for compilers that don't handle inline constants well. There are ports that I don't know about or can't test, and there are some outside the main jdk/jdk tree. If this change breaks any of them, all they have to do is define USE_EXTERN_CONST_FOR_REGISTER_DEFINITIONS. If it turns out that no such systems still exist, we can take it out.

Reply:
Are you satisfied with that, Kim? Thanks.

Reply:
Not really.  I don't think we should be retaining dead code to support potential / imaginary compilers.  If such a compiler still exists (and is sufficiently modern to support C++14), the relevant code can be retrieved from the repository.

Reply:
OK, I need to be braver! Done. :-)